iosApp/iosApp/app/AppDelegate.swift

```diff
@@ -39,18 +39,17 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return KotlinUnit()
     }*/
     
+    static func testSponsorSeedFile() throws {
+        try TestsKt.testSponsorSeedFile()
+    }
+    
     func csLog(s:String) -> KotlinUnit{
         CLSLogv(s, getVaList([]))
         return KotlinUnit()
     }
     
     func loadAsset(filePrefix:String, fileType:String) -> String?{
-        do{
-            let bundleFile = Bundle.main.path(forResource: filePrefix, ofType: fileType)
-            return try String(contentsOfFile: bundleFile!)
-        } catch {
-            return nil
-        }
+        return FunctionsKt.loadAssetFromDefault(asset: filePrefix, type: fileType)
```

Comment:
We should discuss this. The point of passing in a Swift lambda is that some code is easier/more flexible coming from Swift rather than relying on kotlin interop. That makes it harder to test, but that's the crux of this task. How do we test code that can't be fully run from kotlin multiplatform shared testing. You *can* write it in Kotlin, but I was avoiding that on purpose. Having said that, if the logic to get the bundle string is 100% Kotlin, loadAsset should go away, as well as the associated lambda in AppContext, and that whole chain should be Kotlin direct. 'loadAsset' only exists as a function arg into AppContex for "staticFileLoader" https://github.com/touchlab/DroidconKotlin/blob/master/iosApp/iosApp/app/AppDelegate.swift#L23

Reply:
loadAssetFromDefault function is 100% kotlin. Also, test logic is Kotlin in commonMain and respective platform tests invoke the common test logic functions.

I think I'm for replacing staticFileLoader with loadAssetFromDefault calls but not sure of deeper implications with regards to Multiplatform.